home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/CooperationServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+package com.softserveinc.ita.homeproject.homeservice.service.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import com.softserveinc.ita.homeproject.homedata.entity.Cooperation;
+import com.softserveinc.ita.homeproject.homedata.entity.House;
+import com.softserveinc.ita.homeproject.homedata.repository.CooperationRepository;
+import com.softserveinc.ita.homeproject.homedata.repository.HouseRepository;
+import com.softserveinc.ita.homeproject.homeservice.dto.CooperationDto;
+import com.softserveinc.ita.homeproject.homeservice.exception.NotFoundHomeException;
+import com.softserveinc.ita.homeproject.homeservice.mapper.ServiceMapper;
+import com.softserveinc.ita.homeproject.homeservice.service.CooperationService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CooperationServiceImpl implements CooperationService {
+
+    private static final String NOT_FOUND_COOPERATION = "Can't find cooperation with given ID:";
+
+    private static final String FORMAT = "%s %d";
```

Comment:
What's going on here? 

Why cannot we combine `NOT_FOUND_COOPERATION` with `FORMAT`?

So we'll have single constant with a message format:
```java
private static final String NOT_FOUND_COOPERATION_FORMAT = "Can't find cooperation with given ID: %d";
```